nexus/resource_repository_yum_hosted.go

```diff
@@ -0,0 +1,303 @@
+)
+
+func resourceRepositoryYumHosted() *schema.Resource {
+	return &schema.Resource{
+		Create: resourceYumRepositoryCreate,
+		Read:   resourceYumRepositoryRead,
+		Update: resourceYumRepositoryUpdate,
+		Delete: resourceYumRepositoryDelete,
+		Exists: resourceYumRepositoryExists,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Description: "A unique identifier for this repository",
+				Required:    true,
+				Type:        schema.TypeString,
+			},
+			"online": {
+				Default:     true,
+				Description: "Whether this repository accepts incoming requests",
+				Optional:    true,
+				Type:        schema.TypeBool,
+			},
+			"repodata_depth": {
+				Default:     0,
+				Description: "Specifies the repository depth where repodata folder(s) are created. Possible values: 0-5",
+				Optional:    true,
+				Type:        schema.TypeInt,
+			},
+			"deploy_policy": {
+				Default:     "STRICT",
+				Description: "Validate that all paths are RPMs or yum metadata. Possible values: `STRICT` or `PERMISSIVE`",
+				Optional:    true,
+				Type:        schema.TypeString,
+			},
+			"cleanup": {
+				Description: "Cleanup policies",
+				Type:        schema.TypeList,
+				Computed:    true,
+				Optional:    true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"policy_names": {
+							Description: "List of policy names",
+							Elem: &schema.Schema{
+								Type: schema.TypeString,
+							},
+							Optional: true,
+							Set: func(v interface{}) int {
+								return schema.HashString(strings.ToLower(v.(string)))
+							},
+							Type: schema.TypeSet,
+						},
+					},
+				},
+			},
+			"storage": {
+				Description: "The storage configuration of the repository",
+				DefaultFunc: repositoryStorageDefault,
+				Type:        schema.TypeList,
+				Optional:    true,
+				Computed:    true,
+				MaxItems:    1,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"blob_store_name": {
+							Default:     "default",
+							Description: "Blob store used to store repository contents",
+							Optional:    true,
+							Type:        schema.TypeString,
+						},
+						"strict_content_type_validation": {
+							Default:     true,
+							Description: "Whether to validate uploaded content's MIME type appropriate for the repository format",
+							Optional:    true,
+							Type:        schema.TypeBool,
+						},
+						"write_policy": {
+							Description: "Controls if deployments of and updates to assets are allowed",
+							Default:     "ALLOW",
+							Optional:    true,
+							Type:        schema.TypeString,
+							ValidateFunc: validation.StringInSlice([]string{
+								"ALLOW",
+								"ALLOW_ONCE",
+								"DENY",
+							}, false),
+						},
+					},
+				},
+			},
+			/* This is remains here until the go client has been adpated */
+			"yum": {
```

Comment:
can you handle this in the CRUD functions so it can be deleted?

Reply:
No, I tried ... failed hard.

Reply:
I tried again ... made it.